kube: Bump resource versions to what kube 1.16 wants

mook-as opened the pull request:
- `extensions`/`v1beta1/Deployment` moved to `apps/v1`
- `extensions/v1beta1`/`PodSecurityPolicy` moved to `policy/v1beta1`
- `apps/v1beta1`/`StatefulSet` moved to `apps/v1`
- Add support for fallback versions (to run on older kubes), even though we probably no longer support those obsolete kube clusters.

Fixes #514

mook-as committed:
kube: Bump resource versions to what kube 1.16 wants

- extensions/v1beta1/Deployment moved to apps/v1
- extensions/v1beta1/PodSecurityPolicy moved to policy/v1beta1
- apps/v1beta1/StatfulSet moved to apps/v1
- Add support for fallback versions (to run on older kubes), even though
  we probably no longer support those obsolete kube clusters.

Fixes #514

jandubois commented on kube/utils.go:
```diff
@@ -75,6 +75,23 @@ func (b *ConfigBuilder) SetAPIVersion(apiVersion string) *ConfigBuilder {
+		b.apiVersion = apiVersion
+	} else {
+		b.apiVersion = fmt.Sprintf(
+			`{{ ternary "%s" "%s" (.Capabilities.APIVersions.Has "%s") }}`,
```
Do we document anywhere the minimum helm version required? I think `ternary` requires helm 2.10 (which seems sensible, even though we sometimes use 2.8.x).

Should we have a test in the generated charts to make sure the helm version is compatible?

mook-as replied:
Looks like we're still testing against [2.8.2](https://github.com/SUSE/cf-ci/blob/develop/cf-ci-orchestration/Dockerfile#L41), hmm.

mook-as replied:
Okay, so it turns out `helm template` does _not_ support `.Capabilities.APIVersions.Has` style checks, so we can't just run that to see if we have support for older clients.

Will need to get rid of `ternary`, I guess. (But yes, that doesn't work, though a good document on what versions we support, and figuring out how to test them, would be nice)

mook-as replied:
Removed `ternary` in c642ece2c6e51fbd825c2e371a924a09ecdefcd4. Kind of annoyed that I can't actually test with `helm` in a test, though.

mook-as committed:
kube/pod_test.go: Support golang 1.13

golang 1.13 changed the error message emitted when the port in a URL is not
a number; update the test to accept both the old and new messages so that
the test can pass consistently.

mook-as committed:
kube/utils: Drop use of ternrary function

It's not supported in helm 2.8.2, which it appears that we still support.

jandubois reviewed (approved):
Looks good to me (but I didn't actually test it on a kube 1.16 system)
